samples/pyarrow/append_rows_with_arrow.py

```diff
@@ -0,0 +1,206 @@
+        ),
+    ]
+)
+
+
+def bqstorage_write_client():
+    from google.cloud import bigquery_storage_v1
+
+    return bigquery_storage_v1.BigQueryWriteClient()
+
+
+def make_table(project_id, dataset_id, bq_client):
+    table_id = "append_rows_w_arrow_test"
+    table_id_full = f"{project_id}.{dataset_id}.{table_id}"
+    bq_table = bigquery.Table(table_id_full, schema=BQ_SCHEMA)
+    created_table = bq_client.create_table(bq_table)
+
+    return created_table
+
+
+def create_stream(bqstorage_write_client, table):
+    stream_name = f"projects/{table.project}/datasets/{table.dataset_id}/tables/{table.table_id}/_default"
+    request_template = gapic_types.AppendRowsRequest()
+    request_template.write_stream = stream_name
+
+    # Add schema to the template.
+    arrow_data = gapic_types.AppendRowsRequest.ArrowData()
+    arrow_data.writer_schema.serialized_schema = PYARROW_SCHEMA.serialize().to_pybytes()
+    request_template.arrow_rows = arrow_data
+
+    append_rows_stream = AppendRowsStream(
+        bqstorage_write_client,
+        request_template,
+    )
+    return append_rows_stream
+
+
+def generate_write_requests(num_rows=TABLE_LENGTH):
+    date_1 = datetime.date(2020, 10, 1)
+    date_2 = datetime.date(2021, 10, 1)
+
+    datetime_1 = datetime.datetime(2016, 12, 3, 14, 11, 27, 123456)
+    datetime_2 = datetime.datetime(2017, 12, 3, 14, 11, 27, 123456)
+
+    timestamp_1 = datetime.datetime(
+        1999, 12, 31, 23, 59, 59, 999999, tzinfo=datetime.timezone.utc
+    )
+    timestamp_2 = datetime.datetime(
+        2000, 12, 31, 23, 59, 59, 999999, tzinfo=datetime.timezone.utc
+    )
+
+    # Pandas Dataframe.
+    rows = []
+    for i in range(num_rows):
+        row = {
+            "bool_col": True,
+            "int64_col": i,
+            "float64_col": float(i),
+            "numeric_col": decimal.Decimal("0.000000001"),
+            "bignumeric_col": decimal.Decimal("0.1234567891"),
+            "string_col": "data as string",
+            "bytes_col": str.encode("data in bytes"),
+            "date_col": datetime.date(2019, 5, 10),
+            "datetime_col": datetime_1,
+            "time_col": datetime.time(23, 59, 59, 999999),
+            "timestamp_col": timestamp_1,
+            "geography_col": "POINT(-121 41)",
+            "range_date_col": {"start": date_1, "end": date_2},
+            "range_datetime_col": {"start": datetime_1, "end": datetime_2},
+            "range_timestamp_col": {"start": timestamp_1, "end": timestamp_2},
+        }
+        rows.append(row)
+    df = pd.DataFrame(rows)
+
+    # Dataframe to PyArrow Table.
+    table = pa.Table.from_pandas(df, schema=PYARROW_SCHEMA)
+
+    # Determine max_chunksize of the record batches. Because max size of
+    # AppendRowsRequest is 10 MB, we need to split the table if it's too big.
+    # See: https://github.com/googleapis/googleapis/blob/27296636cf8797026124cd67034b42190ab602a4/google/cloud/bigquery/storage/v1/storage.proto#L422
```

Comment:
A more general link is probably preferable here, like https://cloud.google.com/bigquery/docs/reference/storage/rpc/google.cloud.bigquery.storage.v1#appendrowsrequest

Reply:
done